src/requirements/requirement-graph-builder-from-user-data.ts

```diff
@@ -99,7 +106,113 @@ export default function buildRequirementFulfillmentGraphFromUserData(
+    toggleableRequirementChoices,
+    {}, // Provide no double counting choices, so all the edges will be kept
+    overriddenFulfillmentChoices,
+    /* keepCoursesWithoutDoubleCountingEliminationChoice */ true
+  );
+
+  // In this graph, each course is only connected to the selected requirement and the requirements
+  // that allow double counting.
+  const graphWithDoubleCountingAccounted = buildRequirementFulfillmentGraphFromUserData(
+    coursesTaken,
+    onboardingData,
+    toggleableRequirementChoices,
+    selectableRequirementChoices,
+    overriddenFulfillmentChoices,
+    /* keepCoursesWithoutDoubleCountingEliminationChoice */ false
+  ).requirementFulfillmentGraph;
+
+  /**
+   * The table below summerizes the type of all possible requirement-course edges before
+   * double-counting elimination, and where they will go in the new format.
+   *
+   * -------------------------------------------------------------------------------------------------
+   * | Requirement Type | edge exists after double-counting elim | where is it in the new format     |
+   * | ---------------- | -------------------------------------- | --------------------------------- |
+   * | selected         | True                                   | implicit (connected by default)   |
+   * | (no warning)     |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | selected         | True                                   | `acknowledgedCheckerWarningOptIn` |
+   * | (has warning)    |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | allow double     | True                                   | implicit (connected by default)   |
+   * | counting         |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | not connected    | False                                  | `optOut`                          |
+   * | (no warning)     |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | not connected    | False                                  | implicit (unconnected by default) |
+   * | (has warning)    |                                        |                                   |
+   */
+  return function builder(course) {
+    const connectedRequirementsWithDoubleCountingAccounted = graphWithDoubleCountingAccounted.getConnectedRequirementsFromCourse(
+      course
+    );
+    const connectedRequirementsWithoutDoubleCountingAccounted = graphWithoutDoubleCountingAccounted.getConnectedRequirementsFromCourse(
+      course
+    );
```

Comment:
I think it might be more readable if some of these variable and function names were shorter